src/Aspire.Cli/Backchannel/BackchannelDataTypes.cs

```diff
@@ -68,78 +21,6 @@ internal sealed class DisplayLineState(string stream, string line)
     public string Line { get; } = line;
 }
 
-/// <summary>
-/// Envelope for publishing activity items sent over the backchannel.
-/// </summary>
-internal sealed class PublishingActivity
-{
-    /// <summary>
-    /// Gets the type discriminator for the publishing activity item.
-    /// </summary>
-    public required string Type { get; init; }
-
-    /// <summary>
-    /// Gets the data containing all properties for the publishing activity item.
-    /// </summary>
-    public required PublishingActivityData Data { get; init; }
-}
-
-/// <summary>
-/// Common data for all publishing activity items.
-/// </summary>
-internal sealed class PublishingActivityData
-{
-    /// <summary>
-    /// Gets the unique identifier for the publishing activity.
-    /// </summary>
-    public required string Id { get; init; }
-
-    /// <summary>
-    /// Gets the status text describing the publishing activity.
-    /// </summary>
-    public required string StatusText { get; init; }
-
-    /// <summary>
-    /// Gets the completion state of the publishing activity.
-    /// </summary>
-    public string CompletionState { get; init; } = CompletionStates.InProgress;
-
-    /// <summary>
-    /// Gets a value indicating whether the publishing activity is complete.
-    /// </summary>
-    public bool IsComplete => CompletionState is not CompletionStates.InProgress;
-
-    /// <summary>
-    /// Gets a value indicating whether the publishing activity encountered an error.
-    /// </summary>
-    public bool IsError => CompletionState is CompletionStates.CompletedWithError;
-
-    /// <summary>
-    /// Gets a value indicating whether the publishing activity completed with warnings.
-    /// </summary>
-    public bool IsWarning => CompletionState is CompletionStates.CompletedWithWarning;
-
-    /// <summary>
-    /// Gets the identifier of the step this task belongs to (only applicable for tasks).
-    /// </summary>
-    public string? StepId { get; init; }
-
-    /// <summary>
-    /// Gets the completion message for the publishing activity (optional).
-    /// </summary>
-    public string? CompletionMessage { get; init; }
-}
-
-/// <summary>
-/// Constants for publishing activity item types.
-/// </summary>
-internal static class PublishingActivityTypes
-{
-    public const string Step = "step";
-    public const string Task = "task";
-    public const string PublishComplete = "publish-complete";
-}
-
 internal class BackchannelLogEntry
```

Comment:
This one can be deduped. 

https://github.com/dotnet/aspire/blob/2f034511a13cc08e3614dc5f242579516163ce60/src/Aspire.Hosting/Backchannel/BackchannelLoggerProvider.cs#L92-L99

Reply:
I was wondering why this one wasn't shared...